### PR TITLE
Updated eslint and associated packages

### PR DESCRIPTION
**Pull request by mhuggins**
## Description
This cherry-picks the eslint-related greenkeeper PR's into a single PR where the overall changes can be seen in unison.

## Motivation and Context
Greenkeeper had various eslint-related package updates, but it was hard to see the effect of each individual within our client codebase.  This helps to understand the impact of the change so that we can override rules and update syntax as necessary.

## How Has This Been Tested?
`yarn lint`

## Types of changes
- [x] Chore (tests, refactors, and fixes)
- [ ] New feature (adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] I have read the **CONTRIBUTING** document.

## Closing issues
N/A

### CODE REVIEW

File: src/renderer/shared/hocs/withProgressChange.js [hunk 1]
```diff
@@ -21,8 +21,10 @@ export default function withProgressChange(actions, progress, callback) {
   return (Component) => {
     class WrappedComponent extends React.Component {
       componentWillReceiveProps(nextProps) {
-        if (!progresses.includes(this.props[PROGRESS_PROP])
-            && progresses.includes(nextProps[PROGRESS_PROP])) {
+        if (
+          !progresses.includes(this.props[PROGRESS_PROP]) &&
```
Comment (DalderupMaurice) [inline src/renderer/shared/hocs/withProgressChange.js]: I'm not a big fan of this if statement xD

Comment (mhuggins) [reply src/renderer/shared/hocs/withProgressChange.js]: @DalderupMaurice I reverted this, other than the `&&` placement.

File: src/renderer/shared/components/Forms/Button/Button.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import styles from './Button.scss';
 
 export default class Button extends React.Component {
   render() {
-    return (
+    return ( // eslint-disable-next-line react/button-has-type
```
Comment (DalderupMaurice) [inline src/renderer/shared/components/Forms/Button/Button.js]: could you add in the type here as well, just like the other buttons? :P

**Comment by DalderupMaurice** [comment]
What rule was causing the if-else to be so weird?

**Comment by mhuggins** [comment]
I just thought it looked better. 🤷‍♂️

File: src/renderer/shared/components/Forms/Button/Button.js (hunk 2 above)
Comment (mhuggins) [reply src/renderer/shared/components/Forms/Button/Button.js]: The type is in here implicitly.  It's part of the `props`, which we splat here with the spread operator.  If you check out the `defaultProps` further down in this file, you can see it's set to "button".

Comment (mhuggins) [reply src/renderer/shared/components/Forms/Button/Button.js]: eslint doesn't recognize it though, which is why I added the `eslint-disable-next-line` comment here.

**Pull request merged**